From 47fc83cb371a9cc8a8c058097de5e0298cd548f5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robert=20Sch=C3=BCtz?= <rschuetz17@gmail.com>
Date: Sat, 27 Mar 2021 09:29:22 +0100
Subject: [PATCH] [BUG FIX] Use CMAKE_INSTALL_FULL_<dir> in arpack.pc (#301)

CMAKE_INSTALL_<dir> cannot be assumed to be relative to CMAKE_INSTALL_PREFIX.
See https://github.com/jtojnar/cmake-snips#assuming-cmake_install_dir-is-relative-path.
---
 CHANGES        | 3 ++-
 CMakeLists.txt | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0bda0b3..c9d23f45 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -681,8 +681,8 @@ include(GNUInstallDirs)
 # Convert variable names to those expected by the .pc file.
 set(prefix ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix \${prefix})
-set(libdir \${exec_prefix}/${CMAKE_INSTALL_LIBDIR})
-set(includedir \${prefix}/${CMAKE_INSTALL_INCLUDEDIR})
+set(libdir ${CMAKE_INSTALL_FULL_LIBDIR})
+set(includedir ${CMAKE_INSTALL_FULL_INCLUDEDIR})
 set(PACKAGE_NAME ${PROJECT_NAME})
 set(PACKAGE_VERSION ${arpack_ng_VERSION})
 set(PACKAGE_URL "https://github.com/opencollab/arpack-ng/")
